moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -1408,5 +1408,50 @@ void RobotModel::computeFixedTransforms(const LinkModel* link, const Eigen::Isom
                              transform * link->getJointOriginTransform(), associated_transforms);
 }
 
+// Calculating Hash for RobotModel
+namespace std
+{
+	template<> struct hash<RobotModel>
+	{
+		std::size_t operator()(RobotModel const& r) const noexcept 
+    {
+			std::size_t h =0;
+      boost::hash_combine(h, r.model_name_); 
+      boost::hash_combine(h, r.model_frame_);
+
+
+      /**
+       * RobotModel consists of 2: Joint and Link
+       * Additionally it has 1) Indexing for Joint and Link 2) Groups
+       * Groups would be needed for EndEffectors which comes from SRDF (yet to look at JointModelGroups)
+       */
+
+      for (LinkModel* link_model : link_model_vector_) {
+        boost::hash_combine(h, std::hash<LinkModel>{}link_model->hash());
+      }
+
+      for (JointModel* joint_model : joint_model_vector_) {
```

Comment:
:+1: These are sorted... so there should not be a need to use `joint_model_map_`.